src/main/java/org/rumbledb/compiler/StaticContextVisitor.java

```diff
@@ -226,9 +226,9 @@ public StaticContext visitCountClause(CountClause expression, StaticContext argu
     private StaticContext visitFlowrVarDeclaration(FlworVarDecl expression, StaticContext argument) {
         StaticContext result = new StaticContext(argument);
         // TODO for now we only suppot as/default, no inference, flags
-        SequenceType type = expression.getAsSequence() == null
-            ? new SequenceType()
-            : expression.getAsSequence().getSequence();
+        SequenceType type = expression.getSequenceType() == null
+            ? SequenceType.emptySequence
```

Comment:
I love the use of static class field instead of a magic value here.

Reply:
Now I made sure sequence types are never null (this was a mistake in the translator that was incorrectly setting them to null rather than item*).

Reply:
I am very glad that we have clarified that and the code already looks much nicer. I believe there may be a few spots where sequenceType can still be null in an expression. I will mark these with comments. Once we ensure that TranslationVisitor assigns every sequenceType instance to a non-null value. I believe the need for the 'type' variable will go away in this 'StaticContextVisitor.java' file at lines 229 and 265.

Reply:
Yes, my last commit fixes this and makes sure sequenceType is always initiated in the translator.